### PR TITLE
Add support for the GDS pipeline catalog

### PR DESCRIPTION
**Pull request by adamnsch**

### CODE REVIEW

**Comment by FlorentinD** [comment]
this PR failes as it depends on changes in https://github.com/neo-technology/graph-analytics/pull/4796

File: README.md [hunk 1]
```diff
@@ -212,6 +212,23 @@ Further, there's a new call named `gds.graph.get` (`graphdatascience` only) whic
...
+
+```python
+res = gds.beta.pipeline.list()
+assert len(res) == 1  # Exactly one pipeline is in the catalog
+
+res = gds.beta.pipeline.drop(pipe)
+assert res["pipelineName"] == pipe.name()
+```
+
+Further, there's a new call named `gds.pipeline.get` (`graphdatascience` only) which takes a pipeline name as input and returns a training pipeline object if a pipeline of that name exists in the user's pipeline catalog.
```
Comment (jjaderberg) [inline README.md:228]: This is the readme, which doesn't have a contextual dimension of time. Maybe we should avoid saying "new" here?

Comment (jjaderberg) [inline README.md:228]: Also suggest breaking this sentence into two, for readability

```suggestion
Further, there's a call named `gds.pipeline.get` (`graphdatascience` only).
It takes a pipeline name as input and returns a training pipeline object, if a pipeline of that name exists in the user's pipeline catalog.
```

File: graphdatascience/model/model.py [hunk 2]
```diff
@@ -55,3 +56,24 @@ def drop(self) -> None:
...
+    def _query_prefix(self) -> str:
+        pass
+
+    def metrics(self) -> Dict[str, Any]:
+        return self._list_info()["modelInfo"]["metrics"]  # type: ignore
+
+    def predict_stream(self, G: Graph, **config: Any) -> QueryResult:
+        query = f"{self._query_prefix()}stream($graph_name, $config)"
+        config["modelName"] = self.name()
+        params = {"graph_name": G.name(), "config": config}
```
Comment (jjaderberg) [inline graphdatascience/model/model.py:70]: it's a bit unfortunate to have to mix snake case and camel case conventions. is the user required to do that also, or are we making it possible for them to stay "pythonic" all the way?

**Review by jjaderberg — APPROVED** [review]
Looks good!
Except for the failing tests :-)

File: README.md (hunk 1 above)
Comment (adamnsch) [reply README.md:228]: Good point :)

Comment (adamnsch) [reply README.md]: Done

**Pull request merged**